pkg/webhooks/kafkatopic_validator.go

```diff
@@ -154,37 +161,61 @@ func (s *KafkaTopicValidator) checkKafka(ctx context.Context, topic *banzaicloud
 		return nil, errors.WrapIff(err, fmt.Sprintf("failed to list topics for kafka cluster: %s", topic.Spec.ClusterRef.Name))
 	}
 
+	var allErrs field.ErrorList
 	// The topic exists
 	if existing != nil {
 		// Check if this is the correct CR for this topic
 		topicCR := &banzaicloudv1alpha1.KafkaTopic{}
 		if err := s.Client.Get(ctx, types.NamespacedName{Name: topic.Name, Namespace: topic.Namespace}, topicCR); err != nil {
+			// Checking that the validation request is update
 			if apierrors.IsNotFound(err) {
-				// User is trying to overwrite an existing topic - bad user
-				logMsg := fmt.Sprintf("topic already exists on kafka cluster '%s'", topic.Spec.ClusterRef.Name)
-				return field.Invalid(field.NewPath("spec").Child("name"), topic.Spec.Name, logMsg), nil
+				if manager, ok := topic.GetAnnotations()[TopicManagedByAnnotationKey]; !ok || strings.ToLower(manager) != TopicManagedByKoperatorAnnotationValue {
```

Comment:
**Recommended**: We **SHOULD** possibly move the `isTopicManagedByKoperator` here and then we could reuse it here for branching so we wouldn't need to duplicate the condition.

Reply:
This has to be different what is in the isTopicManagedByKoperator function.
In this case the topic already exist so when there is no managedBy annotation or the managedBy annotation is not Koperator then we reject this kafkatopic because the kafka topic is exists and need confirmation.

In the kafkatopic controller: when there is no managedBy annotation or the annotation is koperator then the manager is koperator.

Reply:
Yeah, it makes sense with the `!ok` clearup, thanks.